File: classes/201-w24/Hashing-Lab/HashTable.java
/**
 * A hash table starter class to be used in the Hashing lab.
 * @author Anya Vostinar
 */
public class HashTable {
  
    private Integer[] hashTable;

    /**
     * Constructor for the hash table
     * @param size the size of the hash table
     */
    public HashTable(int size) {
      hashTable = new Integer[size];
    }
  
    /**
    * Compression function that takes a hash code (positive or negative)  and compresses
    * the hash code into the range [0, sizeOfhashTable).
    * @param hashCode the hash code
    * @return the compressed value
    */
    private int compressToSize(int hashCode) {
      int numberOfBuckets = hashTable.length;
      int compressedValue = hashCode % numberOfBuckets;
      if(compressedValue >= 0) {
        return compressedValue;
      } else {
        return compressedValue+numberOfBuckets;
      }
    }

    /**
     * Creates string representation of the hash table
     * @return string 
     */
    public String toString(){
      String returnString = "";
      for(int i = 0; i<hashTable.length; i++){
          returnString += hashTable[i] + " ";
      }
      return returnString;
  }

  /** 
   * Simple insertion method that ignores what is already there and just 
   * inserts the key into the hash table
   * @param key the key to insert
   */
  public void insertKey(Integer key) {
    int pos = compressToSize(key);
    hashTable[pos] = key;
  }
  
    public static void main(String[] args) {
      HashTable table = new HashTable(10);
      table.insertKey(56);
      System.out.println(table);
      
    }
  }

File: classes/201-f25/HashTable.kt
/**
 * In class lab for learning about hash code functions and collisions
 * Based on Java lab by Anna Rafferty
 * Translated to Kotlin and updated by Anya Vostinar
 */


class HashTable {
    var tableSize = 10
    //array of nulls to hold keys
    var keys: Array<String?> = arrayOfNulls<String>(tableSize)


    /**
     * A hashing function that uses the ASCII value of the first character of a string
     * @param s the string to hash
     * @return the hash code
     */
    fun hashCode1(s: String) : Int {
        //TODO
        return 0
    }

    /**
     * A hashing function that uses the sum of the ASCII values of the characters in a string.
     * @param s the string to hash
     * @return the hash code
     */
    fun hashCode2(s: String) : Int {
        //TODO
        return 0
    }

    /**
     * A hashing function that uses the sum of the ASCII values of the characters in a string,
     * weighted by the position of the character in the string.
     * @param s the string to hash
     * @return the hash code
     */
    fun hashCode3(s: String) : Int {
       //TODO
       return 0 
    }

    fun compressToSize(hashCode: Int) : Int {
        var compressedValue = hashCode % tableSize
        if(compressedValue >= 0) {
            return compressedValue
        } else {
            return compressedValue+tableSize
            //Mod of a negative number is negative - 
            //map back to positive so in the range of 
            //[0,numberOfBuckets)
        }
    }

    fun insertKey(str : String) {
        //TODO
    }

    override fun toString() : String {
        var returnString = ""
        for(key in keys) {
            returnString += key + " "
        }
        return returnString
    }

}

fun main() {
    var test = HashTable()
    println(test)

}